test/e2e/specs/editor/various/block-bindings.spec.js

```diff
@@ -0,0 +1,1002 @@
+					args: { key: 'text_custom_field' },
+				},
+			},
+		},
+	},
+};
+const multipleBindingsImageBlock = {
+	name: 'core/image',
+	attributes: {
+		url: imagePlaceholder,
+		alt: 'a',
+		title: 't',
+		metadata: {
+			bindings: {
+				url: {
+					source: 'core/post-meta',
+					args: { key: 'url_custom_field' },
+				},
+				alt: {
+					source: 'core/post-meta',
+					args: { key: 'text_custom_field' },
+				},
+			},
+		},
+	},
+};
+// Image variables.
+const imageReplaceName = 'Replace';
+const imageAltLabel = 'Alternative text';
+const imageTitleLabel = 'Title attribute';
+
+test.describe( 'Block bindings - Post/page context', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'emptytheme' );
+	} );
+
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllPosts();
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'twentytwentyone' );
+	} );
+
+	// Paragraph block tests.
+	test( 'Paragraph - should show the value of the custom field when exists', async ( {
+		editor,
+	} ) => {
+		await editor.insertBlock( contentBindingParagraphBlock );
+		const paragraphBlock = editor.canvas.getByRole( 'document', {
+			name: 'Block: Paragraph',
+		} );
+		const paragraphContent = await paragraphBlock.textContent();
+		expect( paragraphContent ).toBe( textCustomFieldValue );
+	} );
+
+	test( "Paragraph - should show the value of the key when custom field doesn't exists", async ( {
+		editor,
+	} ) => {
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: {
+				content: 'p',
+				metadata: {
+					bindings: {
+						content: {
+							source: 'core/post-meta',
+							args: { key: 'non_existing_custom_field' },
+						},
+					},
+				},
+			},
+		} );
+		const paragraphBlock = editor.canvas.getByRole( 'document', {
+			name: 'Block: Paragraph',
+		} );
+		const paragraphContent = await paragraphBlock.textContent();
+		expect( paragraphContent ).toBe( 'non_existing_custom_field' );
+	} );
+
+	test( 'Paragraph - should lock the appropriate controls', async ( {
+		editor,
+		page,
+	} ) => {
+		await editor.insertBlock( contentBindingParagraphBlock );
+		const paragraphBlock = editor.canvas.getByRole( 'document', {
+			name: 'Block: Paragraph',
+		} );
+		await paragraphBlock.click();
+
+		// Alignment controls exist.
+		await expect(
+			page.getByRole( 'button', {
+				name: alignName,
+			} )
+		).toBeVisible();
+
+		// Format controls don't exist.
+		await expect(
+			page.getByRole( 'button', {
+				name: boldName,
+			} )
+		).toBeHidden();
+
+		// Paragraph is not editable.
+		const isContentEditable =
+			await paragraphBlock.getAttribute( 'contenteditable' );
+		expect( isContentEditable ).toBe( 'false' );
+	} );
+
+	// Heading block tests.
+	test( 'Heading - should show the value of the custom field', async ( {
```

Comment:
You can add a `test.describe( 'Heading', ...`) call to group tests related to a specific block instead of adding prefixes everywhere.

Reply:
I've group the tests in [this commit](https://github.com/WordPress/gutenberg/pull/58550/commits/f6a0a74f7f67370955670d747981c7caf2cf2230).